module/executiondatasync/pruner/pruner.go

```diff
@@ -169,9 +168,16 @@ func (p *Pruner) loop(ctx irrecoverable.SignalerContext, ready component.ReadyFu
 		select {
 		case <-ctx.Done():
 			return
-		case height := <-p.fulfilledHeights:
-			if height > p.lastFulfilledHeight {
-				p.lastFulfilledHeight = height
+		case <-p.fulfilledHeight.Channel():
+			if len(p.registeredProducers) > 0 {
+				lowestHeight := p.lowestProducersHeight()
+
+				err := p.updateFulfilledHeight(lowestHeight)
+				if err != nil {
+					p.logger.Fatal().Err(err).Msg("failed to set fulfilled height")
+					//TODO: check if Throw is needed here
+					//ctx.Throw(fmt.Errorf("failed to update lowest fulfilled height: %w", err))
```

Comment:
yes, this should throw instead of logging a Fatal here.

Reply:
[4b87f42](https://github.com/onflow/flow-go/pull/6217/commits/4b87f42a1f9693edeb7cfcc1c959532202400d71)